source/lac/petsc_parallel_sparse_matrix.cc

```diff
@@ -300,12 +300,19 @@ namespace PETScWrappers
                                               rowstart_in_window.data(),
                                               colnums_in_window.data(),
                                               nullptr);
+          ierr = MatSeqAIJSetPreallocationCSR(matrix,
+                                              rowstart_in_window.data(),
+                                              colnums_in_window.data(),
+                                              nullptr);
```

Comment:
This now calls first `MatMPIAIJSetPreallocationCSR` and then `MatSeqAIJSetPreallocationCSR`, with the same arguments. Is that what you intended? If so, would you mind updating the comment above these two calls to explain why calling both makes sense?

Reply:
Yes, this is how PETSc polymorphism works. An `MPIAIJ` matrix does not implement `SeqAIJSetPreallocationCSR` and vice-versa. So for example, if the matrix is of type `MPIAIJ` the `SeqAIJSet...` call is a dummy call. I will add a comment